### ISSUE TITLE
Keyless dry-run finds default config

### ISSUE THREAD

**Issue by cperciva** [Tarsnap/tarsnap#89]
Not the intended behaviour:
```
$ tarsnap --dry-run --print-stats --humanize-numbers -c /
tarsnap: stat(/root/tarsnap.key): No such file or directory
tarsnap: Cannot read key file: /root/tarsnap.key
```

I think in the command line processing we should record whether a key file or a cache directory were specified *on the command line*; if not (e.g., we have a keyfile set, but it wasn't done on the command line) then we need to add fallback mechanisms for dry runs.

I can't remember the details of cache directories enough to say exactly what needs to be done there without looking at the code, but I think there may also need to be some special "ignore the value which was set in a configuration file" handling there too.

**Comment by gperciva** [comment]
I can't reproduce in an obvious way.  I mean, if I run

    $ tarsnap --dry-run --print-stats --humanize-numbers --no-default-config -c ~/backup

then it's fine.  If I run

    $ tarsnap --dry-run --print-stats --humanize-numbers -c ~/backup

without a `~/.tarsnaprc` then it's fine.  If my `~/.tarsnaprc` is empty (or commented out), then it's also fine.

If `~/.tarsnaprc` points to a non-existent `keyfile` then it complains, but surely that's a PEBCAK ?

Isn't the real problem that somebody ended up with `.../etc/tarsnap.conf` instead of `.../starsnap.conf.sample` ?  and that it wasn't obvious they could add `--no-default-config`?

**Comment by gperciva** [comment]
Thinking "out loud", we don't want to automatically add `--no-default-config` if somebody specifies `--dry-run`, because their config file could usefully specify `--humanize-numbers` and their cache directory.  And we definitely want to use the cache directory if it exists, otherwise the dry run will report that all data would be uploaded (which doesn't change anything in the "pre-signup case", but is definitely not what we want for "post-signup" accounts).

My current thought is that if they specify `--dry-run` and tarsnap encounters an error, it could prompt them to use `--no-default-config` if they haven't signed up.  But the message should be clear that if they *have* signed up, and *have* generated a key, they shouldn't ignore the error.

Also, we should make sure that the tarsnap installation process doesn't end up with an unusable system.  `make install` installs `tarsnap.conf.sample` as such, so maybe we should send a polite request to whichever packager did this?  I've seen at least one few post-install script helpfully rename `tarsnap.conf.sample` to `tarsnap.conf` (to be fair, using tarsnap at all without a keyfile is a new feature).

**Comment by cperciva** [comment]
> Isn't the real problem that somebody ended up with .../etc/tarsnap.conf instead of .../tarsnap.conf.sample ? and that it wasn't obvious they could add --no-default-config?

Yes, but I *want* tarsnap.conf.sample to be installed as tarsnap.conf if no configuration file exists yet; the options in there are recommended.

**Comment by miketweaver** [comment]
Might be a good idea to add --no-default-config to your FAQ section:
https://www.tarsnap.com/faq.html#COST-BEFORE-SIGNUP

**Comment by gperciva** [comment]
Good idea.  I was hoping this would be a quick fix, but even if was, people running 1.0.37 would still be stuck until 1.0.38 came out.

**Comment by gperciva** [comment]
Fixed in #91.
